Team9Project/src/com/indragie/cmput301as1/ExpenseClaimTabFragment.java

```diff
@@ -0,0 +1,482 @@
+				
+				switch (mutation.getMutationType()) {
+				case DELETE:
+					claim.removeTag(tag);
+					if (filteredTagsList.contains(tag) && !filteredTagsList.isEmpty()) {
+						filteredTagsList.remove(tag);
+					}
+					break;
+				case EDIT:
+					int tagIndex = claim.getTags().indexOf(tag);
+					claim.setTag(tagIndex, mutation.getNewTag());
+					if (filteredTagsList.contains(tag) && !filteredTagsList.isEmpty()) {
+						int selectedIndex = filteredTagsList.indexOf(tag);
+						filteredTagsList.set(selectedIndex, mutation.getNewTag());
+					}
+					break;
+				}
+				
+				modifiedClaims.append(i, claim);
+				i++;
+			}
+			
+		}
+		
+		for (int i = 0; i < modifiedClaims.size(); i++) {
+			listModel.set(modifiedClaims.keyAt(i), modifiedClaims.valueAt(i));
+		}
+		if (!filteredTagsList.isEmpty()) {
+			setFilteredClaims();
+		}
+		
+	}
+
+	/**
+	 * Prompts the user for confirmation in response to deleting an expense
+	 * claim.
+	 */
+	public void startDeleteAlertDialog() {
+		AlertDialog.Builder openDialog = new AlertDialog.Builder(activity);
+		openDialog.setTitle(R.string.action_delete_claim_confirm);
+
+		openDialog.setPositiveButton(android.R.string.ok,
+				new DialogInterface.OnClickListener() {
+			@Override
+			public void onClick(DialogInterface dialog, int which) {
+				listModel.remove(longPressedItemIndex);
+
+			}
+		});
+
+		openDialog.setNegativeButton(android.R.string.cancel,
+				new DialogInterface.OnClickListener() {
+			@Override
+			public void onClick(DialogInterface dialog, int which) {
+				dialog.dismiss();
+			}
+		});
+		openDialog.show();
+	}
+	
+	/**
+	 * Checks list of filtered tags. 
+	 * If claim contains filtered tag, we keep the claim in our filteredListModel.
+	 */
+	private void setFilteredClaims() {
+		ArrayList<ExpenseClaim> tempClaims = new ArrayList<ExpenseClaim>();
+		
+		for (Tag tag: filteredTagsList) {
+			for (ExpenseClaim claim: listModel.getItems()) {
+				if (claim.hasTag(tag)) {
+					if (!tempClaims.contains(claim)) {
+						tempClaims.add(claim);
+					}
+				}
+			}
+		}
+		filteredListModel.replace(tempClaims);
+		setListAdapter(new ExpenseClaimArrayAdapter(activity, filteredListModel.getItems()));
+	}
+	
+	/**
+	 * Loads the expense claim data to display in the {@link ListView}
+	 */
+	protected void loadData(int i) {
+		// Create the application-wide session
+		Session session = new Session(activity, userManager.getActiveUser());
+		Session.setSharedSession(session);
+
+		// Show the initial list of expense claims (persisted on disk)
+		listModel = session.getOwnedClaims();
+		listModel.addObserver((TypedObserver<CollectionMutation<ExpenseClaim>>)this);
+		setListAdapter(new ExpenseClaimArrayAdapter(activity, listModel.getItems()));
+
+		//Load the new list from the server
+		final Context context = activity;
+		session.loadOwnedClaims(new ElasticSearchAPIClient.APICallback<List<ExpenseClaim>>() {
+			@Override
+			public void onSuccess(Response response, List<ExpenseClaim> model) {}
```

Comment:
Whoops, you're right. I don't know when that changed, it must've been one of the merges. I'll fix it right now.